### PR TITLE
christis 1.5.4 mega dump

### PR DESCRIPTION
**Pull request by Jonah-Hansen**
1.5.4 is realllll
	1. Remove enig graves, dun gear, soulbound, respawn animals, incendium, terralith, spawn balance utility, fins and tails, goodall, straw golem, inv hud, smooth swapping, just enough advancements, chunk pregenerator
	2. Removed unnecessary and ugly resource packs
	3. Add galosphere, tectonic, void fog, brazier, hts treechop, cluttered, multimine, archaeology api, respawn timer
	4. Disabled biome bosses datapack temporarily again since SBU's removal means they no longer spawn, added back recipes to craft their summons, will make long term solutions soon
	5. Questbook overhaul (temporary version)
	6. Fix controls by setting all vanilla keys to default, and all mod keys to default UNLESS they conflict in which case I changed the less important key to shift or ctrl+
	7. Set default shader options to be potato
	8. Changed loading backgrounds
	9. Fixed parcool once and for all and removed stamina bar since its useless
	10. Fixed roll ui
	11. Buffed netherite monstrositys spd x2, atk x2, def x1.5, ignis atk x2 def x1.5, harbinger x1.5, void worm hp 1.5->x2, spd x1.6, atk *1.6, ender dragon atk x2, wither atk x1.5 spd x1.2, mut wither skelly spd x1.5 atk x1.5 kbr x2, mut blaze spd x1.5 atk x1.5, spiritcaller spd x1.5, magispeller hpx2 atkx1.5, leviathan atk x1.5, rosalyne kbr x1.5, capt cornelia kbr x1.5 hp x1.5 atk x1.5, warped mosco spd x1.5 atk x1.5, mut shulker atk x1.5, corrupted champion atk x1.8 spd x1.5, ender guardian atk x2, prisnautilus spd x1.5 atk x2, colossus spd x1.5, huntsman atk x1.5, queen bee atk x1.5
	12. Updated cataclysm
	13. Fix broken quest items
	14. Removed nightmare stalker from expert mode exclusive >:)
	15. First person model off by default
	16. Reenabled glow caves
	17. Fix menu buttons
	18. Spawn with quest book
	19. Echo/flamarangs break spawners
	20. Buffed max health increase from each food group being 80%+ from 1->2 hearts per group, and an extra 2 if all non sugar groups are 80%+

### CODE REVIEW

File: overrides/scripts/createstages.zs [hunk 1]
```diff
@@ -377,9 +377,7 @@ var itemsCursed = [
 
 
 //Eye gates
-	ItemStages.restrict(<item:dungeons_gear:satchel_of_elixirs>.anyDamage(), "witch_eye").preventInventory(false).preventPickup(false).setHiddenInJEI(false);
```
Comment (GildedToxin) [inline overrides/scripts/createstages.zs:380]: This should be included in the dev build

Comment (GildedToxin) [reply overrides/scripts/createstages.zs:380]: Well, all 3 lines thast seem to be removed

**Comment by Jonah-Hansen** [comment]
confirmed 
1, 2, 3, 4, 
6, 7, 8, 9,10, 
12, 
15, 
17,
19 - regular pickarang does not I assume that is intended.

- only spawned with parcool book, no quest book
- world gen seems bugged. only medium temperature biomes as far as I can tell. couldn't /locate desert or taiga, frozen ocean etc

- quest book open to this page on first open.
image

- most quest/wiki entry points are awarded on spawn which is great, except for "useful hotkeys", "inventory/resource mechanics", "resources" which are dependant on "welcome". "combat" is not dependant but starts out unchecked.

**Comment by GildedToxin** [comment]
I think the QB opens on that page since youre in creative, should be hidden otherwise

**Comment by Jonah-Hansen** [comment]
> I think the QB opens on that page since youre in creative, should be hidden otherwise

nah does it in survival cheats disabled too

**Comment by GildedToxin** [comment]
ah well, nvm then lol

**Comment by GildedToxin** [comment]
What mod list are you using?

**Comment by Jonah-Hansen** [comment]
> What mod list are you using?

the one from the curseforge profile christ made. the one in this repo is not up to date. ill update right after I finish resolving the merge conflicts!

**Comment by GildedToxin** [comment]
5 is good (except for my changes which seem to be getting added with the new commits

Theres an issues with 16 where the *only* biome that spawns (other the plains) is the glow caves (gilmmering wheild), so we should prob turn it off - it happened before in testing with soul and guy

File: overrides/defaultconfigs/majruszsdifficulty-server.toml [hunk 2]
```diff
@@ -297,7 +297,7 @@
...
...n_in_chaos_v1:zombie_clown", "born_in_chaos_v1:mr_pumpkin", "born_in_chaos_v1:sir_pumpkinhead", "born_in_chaos_v1:baby_skeleton_minion", "born_in_chaos_v1:controlled_baby_skeleton", "born_in_chaos_v1:phantom_creeper_copy", "born_in_chaos_v1:bone_imp_minion", "born_in_chaos_v1:supreme_bonescaller_stage_2", "born_in_chaos_v1:spirit_guide_assistant", "born_in_chaos_v1:mr_pumpkin_controlled", "born_in_chaos_v1:felsteed", "born_in_chaos_v1:sir_pumpkinhead_without_horse", "born_in_chaos_v1:sir_the_headless", "born_in_chaos_v1:pumpkinhead", "born_in_chaos_v1:riding_felsteed", "born_in_chaos_v1:seared_spirit_not_despawn", "minecraft:illusioner", "graveyard:acolyte", "graveyard:revenant", "graveyard:ghoul", "graveyard:skeleton_creeper","graveyard:corrupted_pillager", "graveyard:corrupted_vindicator", "graveyard:wraith", "graveyard:reaper", "graveyard:nightmare",  "majruszsdifficulty:tank", "majruszsdifficulty:cerberus", "cataclysm:amethyst_crab", "meetyourfight:rosalyne", "alexsmobs:warped_mosco", "iter_rpg:carcass", "epicsamurai:oni", "epicsamurai:kitsune", "epicsamurai:akaname", "epicsamurai:enenra", "creeperoverhaul:ocean_creeper","born_in_chaos_v1:dire_hound_leader", "alexsmobs:murmur"]
```
Comment (GildedToxin) [inline overrides/defaultconfigs/majruszsdifficulty-server.toml:300]: > "born_in_chaos_v1:nightmare_stalker" seems to be added here properly

Comment (GildedToxin) [reply overrides/defaultconfigs/majruszsdifficulty-server.toml:300]: this means 14 is solved

**Comment by GildedToxin** [comment]
**Confirmed:**
1, 2, 3, 4, 5, 6, 7, 8, 9,10,
12, 13, 14, 15, 
16 - is enabled, but causes all biomes to break, will need fix
17,
19 - regular pickarang does not I assume that is intended.

**Unconfirmed:**
11,
20

**Not working:**
18 - Player does not spawn with QB
16 - is enabled, but causes all biomes to break, will need fix

**Comment by Jonah-Hansen** [comment]
Okay so I synced up this branch with what was in the profile export from christi. @GildedToxin you'll wanna manually delete the config folder in your instance, copy in the config from the 1.5.4 zip, then you can copyToInstance and it'll all be good. main issue was the ftb quest chapters that were deleted in her instance weren't deleted on git.

**Comment by GildedToxin** [comment]
Perfect! Thank you so much!

Once I pull the branch and copy over the configs, would I be good to switch back and forth between branches without having to copy configs over every time?

**Comment by Jonah-Hansen** [comment]
> Once I pull the branch and copy over the configs, would I be good to switch back and forth between branches without having to copy configs over every time?

yeah only reason we have to manually copy config is because we cant have the copy script delete the folder, so stuff that got deleted in development doesn't get removed from the instance.

if I understand you question right though you should be good

File: overrides/scripts/createstages.zs (hunk 1 above)
Comment (Jonah-Hansen) [reply overrides/scripts/createstages.zs:380]: isnt dungeons gear removed so this is fine?

**Comment by Jonah-Hansen** [comment]
> confirmed 1, 2, 3, 4, 6, 7, 8, 9,10, 12, 15, 17, 19 - regular pickarang does not I assume that is intended.
> 
> * only spawned with parcool book, no quest book
> * world gen seems bugged. only medium temperature biomes as far as I can tell. couldn't /locate desert or taiga, frozen ocean etc
> * quest book open to this page on first open.
>   image
> * most quest/wiki entry points are awarded on spawn which is great, except for "useful hotkeys", "inventory/resource mechanics", "resources" which are dependant on "welcome". "combat" is not dependant but starts out unchecked.

I think this might be because the chapter is first in the list? even though its supposed to be op only I assume the first page in the QB is the default for it to open to

**Comment by Jonah-Hansen** [comment]
@RichPrune so all we need fixed before we go ahead and merge this is pr 
18. -> player doesn't spawn with anything besides parcool qb. 
19. 
also the first page it opens to is the temps page. I think just cuz its the first page at order 0 above welcome.

also, most but not all of the quest starting points are awarded when the player joins. would be nice if they all are just for consistency.
> most quest/wiki entry points are awarded on spawn which is great, except for "useful hotkeys", "inventory/resource mechanics", "resources" which are dependant on "welcome". "combat" is not dependant but starts out unchecked.
Not working:

16 ->glow caves. glimmering weald biome seems to be over taking almost all other biomes on world generation. maybe we have to keep it disabled unless there's a way to fix it.

**Comment by GildedToxin** [comment]
> **Confirmed:** 1, 2, 3, 4, 5, 6, 7, 8, 9,10, 12, 13, 14, 15, 16 - is enabled, but causes all biomes to break, will need fix 17, 19 - regular pickarang does not I assume that is intended.
> 
> **Unconfirmed:** 11, 20
> 
> **Not working:** 18 - Player does not spawn with QB 16 - is enabled, but causes all biomes to break, will need fix

fixed 16 and 18

11 and 20 needs testing, but I also trust christi to have it be in there, ill take a glance at the code later, but otherwise I think were good

**Comment by Jonah-Hansen** [comment]
> 11 and 20 needs testing, but I also trust christi to have it be in there, ill take a glance at the code later, but otherwise I think were good

20 was done in datapacks. https://github.com/chocolate-edition/Chocolate-Datapacks/commit/a8531e1b7f16c28c9a64786f3a2eb62d8401cc77

11 has not been implemented on git at least https://github.com/chocolate-edition/Chocolate-Edition/blob/christis-mega-dump/overrides/config/cataclysm.toml

I'll check the profile config and see if the changes are there and copy them over

**Comment by Jonah-Hansen** [comment]
11 is confirmed im just blind lmao its been heeree the whole time

**Comment by GildedToxin** [comment]
lol merge time?

**Review by GildedToxin — APPROVED** [review]

**Comment by Jonah-Hansen** [comment]
LETS GOOOOOOOOOOO

**Pull request merged**